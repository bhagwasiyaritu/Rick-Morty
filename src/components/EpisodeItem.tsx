import {Text, TouchableOpacity} from 'react-native';
import React, {memo} from 'react';
import {EpisodeItemProps} from '../model/characterTypes';

const EpisodeItem: React.FC<EpisodeItemProps> = ({
  id,
  name,
  episode,
  onPress,
}) => {
  return (
    <TouchableOpacity
      key={id}
      className="flex-row border-b-2 border-gray-200 py-4 px-4"
      onPress={onPress}>
      <Text className="flex-1 color-slate-500 font-semibold text-l text-start  ">
        {name}
      </Text>
      <Text className=" color-slate-500 font-semibold text-l text-end ">
        {episode}
      </Text>
    </TouchableOpacity>
  );
};

export default memo(EpisodeItem, (prevProps, nextProps) => {
  return prevProps.id === nextProps.id;
});
