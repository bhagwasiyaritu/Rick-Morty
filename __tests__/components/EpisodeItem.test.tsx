import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import EpisodeItem from '../../src/components/EpisodeItem';

describe('EpisodeItem', () => {
  const mockProps = {
    id: 1,
    name: 'Pilot',
    episode: 'S01E01',
    onPress: jest.fn(),
  };

  it('episode-item renders correctly', () => {
    const {getByText, getByTestId} = render(<EpisodeItem {...mockProps} />);

    expect(getByTestId('episode-item')).toBeTruthy();
    expect(getByText('Pilot')).toBeTruthy();
    expect(getByText('S01E01')).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const {getByTestId} = render(<EpisodeItem {...mockProps} />);

    const episodeItem = getByTestId('episode-item');
    fireEvent.press(episodeItem);

    expect(mockProps.onPress).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const {toJSON} = render(<EpisodeItem {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
