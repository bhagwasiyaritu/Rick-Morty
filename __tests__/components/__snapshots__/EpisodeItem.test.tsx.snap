// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeItem matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="episode-item"
>
  <Text
    className="flex-1 color-slate-500 font-semibold text-l text-start  "
  >
    Pilot
  </Text>
  <Text
    className=" color-slate-500 font-semibold text-l text-end "
  >
    S01E01
  </Text>
</View>
`;
